ui/src/shortcuts/shortcut_event.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt;
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+#[non_exhaustive]
+pub enum ShortcutModifier {
+    Ctrl,
+    Alt,
+    Shift,
+    Super,
+}
+
+impl fmt::Display for ShortcutModifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ShortcutModifier::Ctrl => write!(f, "Ctrl"),
+            ShortcutModifier::Alt => write!(f, "Alt"),
+            ShortcutModifier::Shift => write!(f, "Shift"),
+            ShortcutModifier::Super => write!(f, "Super"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+#[non_exhaustive]
+pub enum ShortcutKey {
+    C,
+    I,
+    R,
+    Q,
+}
+
+impl fmt::Display for ShortcutKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ShortcutKey::C => write!(f, "C"),
+            ShortcutKey::I => write!(f, "I"),
+            ShortcutKey::R => write!(f, "R"),
+            ShortcutKey::Q => write!(f, "Q"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct ShortcutEvent {
+    pub shortcut_modifier: ShortcutModifier,
+    pub shortcut_key: ShortcutKey,
+}
```

Comment:
This is a poor representation of what keys could be pressed, it needs rethinking. It works for now but I would like to see something closer to JS's KeyEvent with a key and optionally modifiers.

This could either be a series of booleans (preferable IMO) _or_ an array of ShortcutModifier.